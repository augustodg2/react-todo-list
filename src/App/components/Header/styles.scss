.header {
  padding: 1rem 1.5rem;
  background: $white;

  h2 {
    font-size: 1.5rem;
    font-weight: 400;
    color: $primary;
  }
}
