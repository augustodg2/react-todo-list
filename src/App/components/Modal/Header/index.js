import React from 'react'
import PropTypes from 'prop-types'
import { MdClose } from 'react-icons/md'
import Button from '../../Button'

const ModalHeader = ({ title, closeButton, onClose }) => {
  return (
    <div className="modal__header">
      { closeButton && (
        <Button
          action={onClose}
          icon={<MdClose size={20} />}
        />
      )}
      <h1 className="modal__title">{title}</h1>
    </div>
  )
}

ModalHeader.propTypes = {
  title: PropTypes.string,
  closeButton: PropTypes.bool,
  onClose: PropTypes.func
}

export default ModalHeader
