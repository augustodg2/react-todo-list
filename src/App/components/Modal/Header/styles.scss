.modal {
  &__header {
    display: flex;

    .button {
      margin-right: .75rem;
    }
  }

  
  &__title {
    font-size: 1rem;
    color: $dark-gray;
    font-weight: 600;
  }
}