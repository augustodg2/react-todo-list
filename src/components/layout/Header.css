.header {
  padding: 1rem 1.5rem;
  
  position: relative;
  box-shadow: 0px 1px 5px #ccc;
  z-index: 2;
  color: #666;

  display: flex;
  align-items: baseline;
}

.header h2 {
  font-size: 1.25rem;
}

.nav-links {
  margin-left: .75rem;
}

.nav-links a {
  color: #ccc;
  display: inline-flex;
  padding: 0 1rem 0 1rem;
}

.nav-links a:hover {
  color: #999;
}