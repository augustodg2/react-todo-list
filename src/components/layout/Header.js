import React from 'react'
import PropTypes from 'prop-types'
import AddTodo from '../AddTodo'

class Header extends React.Component {
  render () {
    return (
      <header style={ headerStyle }>
        <h2>Todo List</h2>
        <small style={ smallStyle }>Developed by Augusto Dias</small>

        <AddTodo addTodo={ this.props.addTodo } />
      </header>
    )
  }
}

const headerStyle = {
  padding: '2rem 1.5rem 1.5rem 1.5rem',
  boxShadow: '0px 1px 5px #ccc',
  position: 'relative',
  zIndex: 2,
  color: '#444'
}

const smallStyle = {
  fontFamily: 'Helvetica Neue',
  color: '#aaa',
  display: 'inline-block',
  paddingBottom: '1.25rem'
}

Header.propTypes = {
  addTodo: PropTypes.func
}

export default Header
