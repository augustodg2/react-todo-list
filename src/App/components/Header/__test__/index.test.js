import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Header from '../index'

afterEach(cleanup)

describe('# Header', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <Header />
    )
    expect(container).toMatchSnapshot()
  })
})
