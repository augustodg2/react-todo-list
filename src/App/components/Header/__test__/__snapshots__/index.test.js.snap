// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Header should match the snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <h2>
      Primary
    </h2>
  </header>
</div>
`;
